kernel/src/config.rs

```diff
@@ -0,0 +1,802 @@
+                    field
+                        .splitn(3, '.')
+                        .last()
+                        .map(|n| Constraint::new(n, value))
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    /// Column names on which Delta Lake collects statistics to enhance data skipping functionality.
+    /// This property takes precedence over [num_indexed_cols](Self::num_indexed_cols).
+    pub fn stats_columns(&self) -> Option<Vec<&str>> {
+        self.0
+            .get(DeltaConfigKey::DataSkippingStatsColumns.as_ref())
+            .map(|v| v.split(',').collect())
+    }
+}
+
+#[derive(
+    Serialize, Deserialize, Debug, Copy, Clone, PartialEq, EnumString, StrumDisplay, AsRefStr,
+)]
+/// The isolation level applied during transaction
+pub enum IsolationLevel {
+    /// The strongest isolation level. It ensures that committed write operations
+    /// and all reads are Serializable. Operations are allowed as long as there
+    /// exists a serial sequence of executing them one-at-a-time that generates
+    /// the same outcome as that seen in the table. For the write operations,
+    /// the serial sequence is exactly the same as that seen in the table’s history.
+    #[strum(ascii_case_insensitive)]
+    Serializable,
+
+    /// A weaker isolation level than Serializable. It ensures only that the write
+    /// operations (that is, not reads) are serializable. However, this is still stronger
+    /// than Snapshot isolation. WriteSerializable is the default isolation level because
+    /// it provides great balance of data consistency and availability for most common operations.
+    #[strum(ascii_case_insensitive)]
+    WriteSerializable,
+
+    /// SnapshotIsolation is a guarantee that all reads made in a transaction will see a consistent
+    /// snapshot of the database (in practice it reads the last committed values that existed at the
+    /// time it started), and the transaction itself will successfully commit only if no updates
+    /// it has made conflict with any concurrent updates made since that snapshot.
+    #[strum(ascii_case_insensitive)]
+    SnapshotIsolation,
+}
+
+// Spark assumes Serializable as default isolation level
+// https://github.com/delta-io/delta/blob/abb171c8401200e7772b27e3be6ea8682528ac72/core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala#L1023
+impl Default for IsolationLevel {
+    fn default() -> Self {
+        Self::Serializable
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, EnumString, StrumDisplay, AsRefStr)]
+/// The checkpoint policy applied when writing checkpoints
+#[serde(rename_all = "camelCase")]
+#[strum(serialize_all = "camelCase")]
+pub enum CheckpointPolicy {
+    /// classic Delta Lake checkpoints
+    Classic,
+    /// v2 checkpoints
+    V2,
+}
+
+impl Default for CheckpointPolicy {
+    fn default() -> Self {
+        Self::Classic
```

Comment:
default to V2 since classic is being deprecated
```suggestion
        Self::V2
```